django/db/backends/oracle/base.py

```
@@ -238,6 +240,38 @@ def __init__(self, *args, **kwargs):
         )
         self.features.can_return_columns_from_insert = use_returning_into
 
+    @property
+    def is_pool(self):
+        return self.settings_dict["OPTIONS"].get("pool")
+
+    @property
+    def pool(self):
+        if not self.settings_dict["OPTIONS"].get("pool"):
+            return None
+
+        # Verify that we are not running with persistent connections
+        if self.settings_dict.get("CONN_MAX_AGE", 0) != 0:
+            raise ImproperlyConfigured(
+                """Pooling doesn't support persistent connections, unset
+                   conn_max_age"""
+            )
+        # Pooling feature is only supported for oracledb
+        if not is_oracledb:
+            raise ImproperlyConfigured("Not supported by cx_Oracle install oracledb")
```

This should raise an exception much earlier when connection params are checked.